public/app/features/alerting/unified/components/rule-editor/PreviewRule.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useCallback, useState } from 'react';
+import { css } from '@emotion/css';
+import { useFormContext } from 'react-hook-form';
+import { Button, HorizontalGroup, useStyles2 } from '@grafana/ui';
+import { dateTimeFormatISO, getDefaultTimeRange, GrafanaTheme2, LoadingState } from '@grafana/data';
+import { RuleFormType } from '../../types/rule-form';
+import { PreviewRuleRequest, PreviewRuleResponse } from '../../types/preview';
+import { previewAlertRule } from '../../api/preview';
+import { PreviewRuleResult } from './PreviewRuleResult';
+import { toDataQueryError } from '@grafana/runtime';
+
+const fields: string[] = ['type', 'dataSourceName', 'condition', 'queries', 'expression'];
+
+export function PreviewRule(): React.ReactElement | null {
+  const [preview, setPreview] = useState<PreviewRuleResponse | undefined>();
+  const styles = useStyles2(getStyles);
+  const { getValues } = useFormContext();
+  const [type] = getValues(fields);
+
+  const onPreview = useCallback(async () => {
```

Comment:
using async code like this in a component risks both race conditions and "set state after unmount" errors. https://github.com/streamich/react-use/blob/master/docs/useAsyncFn.md is perfect to solve both of these